packages/light-apps/src/TopBar/TopBar.tsx

```diff
@@ -2,81 +2,45 @@
 // This software may be modified and distributed under the terms
 // of the Apache-2.0 license. See the LICENSE file for details.
 
-import { BlockNumber, Header } from '@polkadot/types';
-import React from 'react';
-import { Observable, Subscription } from 'rxjs';
+import { Header } from '@polkadot/types';
 import { AppContext } from '@substrate/ui-common';
-import { FadedText, FlexItem, Margin, NavLink, StackedHorizontal } from '@substrate/ui-components';
+import { FadedText, FlexItem, NavLink, StackedHorizontal, Margin } from '@substrate/ui-components';
+import React, { useContext, useEffect, useState } from 'react';
+import { Observable } from 'rxjs';
 
 import { BlockCounter, NodeStatus } from './TopBar.styles';
 import substrateLogo from '@polkadot/ui-assets/parity-substrate.svg';
 
-type State = {
-  blockNumber?: BlockNumber,
-  backupModalOpen: boolean,
-  forgetModalOpen: boolean,
-  error?: string,
-  success?: string,
-  password: string
-};
+interface Props {}
 
-export class TopBar extends React.PureComponent<{}, State> {
-  static contextType = AppContext;
+export function TopBar (props: Props) {
+  const { api, system: { chain, health: { isSyncing }, name, version } } = useContext(AppContext);
 
-  context!: React.ContextType<typeof AppContext>; // http://bit.ly/typescript-and-react-context
+  const [blockNumber, setBlockNumber] = useState();
+  useEffect(() => {
+    const chainHeadSub = (api.rpc.chain.subscribeNewHead() as Observable<Header>)
+      .subscribe((header) => setBlockNumber(header.blockNumber));
 
-  state: State = {
-    backupModalOpen: false,
-    forgetModalOpen: false,
-    password: ''
-  };
+    return () => {
+      chainHeadSub.unsubscribe();
+    };
+  }, [api, setBlockNumber]);
```

Comment:
did you mean to write `blockNumber` here?

Reply:
no, setBlockNumber; otherwise react was warning me that some of the variables I'm using inside the hooks are not referenced in the dependencies